web/server.js

```diff
@@ -225,6 +260,19 @@ module.exports = function(options) {
           {
             assign: 'authCode',
             method: function(request, reply) {
+              if ( request.pre.grant_type === 'password' ) {
+                return account.verifyPassword(request, function(err, json) {
+                  if ( err ) {
+                    return reply(err);
+                  }
+
+                  reply({
+                    user_id: json.user.username,
+                    scopes: request.payload.scopes
```

Comment:
This looks like a red flag to me, blindly allowing any scopes. Is there any way we can limit this? Maybe take the scopes from the oauth client whitelist? I can't think of any other option.